src/reagent/impl/batching.cljs

```diff
@@ -37,7 +37,7 @@
   ;; are rendered before children
   (.sort a compare-mount-order)
   (dotimes [i (alength a)]
-    (let [c (aget a i)]
+    (let [^js/React.Component c (aget a i)]
```

Comment:
I guess this is the trick that fixes the advanced compile bug I see in https://github.com/reagent-project/reagent/issues/449#issuecomment-554625425 ? I wonder how could non-shadow-cljs tools work without this?

Reply:
https://clojurescript.org/guides/externs
https://github.com/cljsjs/packages/blob/master/react/resources/cljsjs/react/common/react.ext.js#L149
https://shadow-cljs.github.io/docs/UsersGuide.html#_why_not_use_cljsjs

Reply:
I see, so other tools would use the externs for react because it's bundled in cljsjs, while shadow-cljs can't make use of it. Thanks for the links!